homeassistant/components/wake_on_lan/switch.py

```diff
@@ -114,31 +118,38 @@ def turn_on(self, **kwargs):
 
         wakeonlan.send_magic_packet(self._mac_address, **service_kwargs)
 
+        if self._is_dummy:
+            self._state = True
```

Comment:
Polling should be disabled so write the state.

```suggestion
            self._state = True
            self.async_write_ha_state()
```